os/user/getent_usefgetent.go

```diff
@@ -0,0 +1,86 @@
+// +build aix dragonfly freebsd !android,linux netbsd openbsd solaris !darwin
+// +build cgo,!osusergo
+
+package user
+
+/*
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+#include <grp.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+
+static void resetErrno(){
+	errno = 0;
+}
+*/
+import "C"
+import "unsafe"
+
+// usefgetpwent is used for testing. Sets getpwent to use fgetpwent.
+func (u *iterateUsersHelper) usefgetpwent(fileName string) error {
+	fname := C.CString(fileName)
+	defer C.free(unsafe.Pointer(fname))
+	mode := C.CString("r")
+	defer C.free(unsafe.Pointer(mode))
+
+	var fp *C.FILE
```

Comment:
this can be shorter:

1. Open the file in go: `f := os.OpenFile(...)`
2. Convert the file descriptor into C's `*FILE`: `fp = C.fopen(fd.Fd(), <...>)`